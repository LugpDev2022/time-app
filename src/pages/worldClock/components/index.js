export * from './AddCountryForm';
export * from './CountryClockItem';
export * from './CountryClockList';
