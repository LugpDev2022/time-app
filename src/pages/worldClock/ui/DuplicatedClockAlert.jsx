import Alert from 'react-bootstrap/Alert';

export const DuplicatedClockAlert = () => (
  <Alert className='mt-3 bg-info text-dark border-dark border fw-bold'>
    You already have that clock
  </Alert>
);
