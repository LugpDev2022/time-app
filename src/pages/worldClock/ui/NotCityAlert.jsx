import React from "react";
import Alert from "react-bootstrap/Alert";

export const NotCityAlert = () => (
  <Alert className="mt-3 bg-info text-dark border-dark border fw-bold">
    Don&apos;t leave the city input empty
  </Alert>
);
