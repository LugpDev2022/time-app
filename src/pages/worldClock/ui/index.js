export * from './DuplicatedClockAlert';
export * from './NotCityAlert';
