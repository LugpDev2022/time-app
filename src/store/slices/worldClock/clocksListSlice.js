import { createSlice } from '@reduxjs/toolkit';

export const clocksListSlice = createSlice({
  //TODO: fetch the api to get the data

  name: 'clocksList',
  initialState: {
    worldClocks: [
      {
        city: 'Quebec',
        date: '00/00/00',
        clock: '00:00',
        id: '123',
      },
    ],
    chargingData: true,
    error: {
      isError: false,
      type: '',
    },
  },

  reducers: {
    addClock: (state, action) => {
      state.worldClocks.push({
        city: action.payload.city,
        date: '00/00/00',
        clock: '00:00',
      });
    },

    handleError: (state, action) => {
      state.error.isError = true;
      state.error.type = action.payload.errorType;
    },

    resetError: state => {
      state.error.isError = false;
      state.error.type = '';
    },
  },
});

export const { addClock, handleError, resetError } = clocksListSlice.actions;
