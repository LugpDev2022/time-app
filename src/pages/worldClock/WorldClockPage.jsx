import { Container, Row } from 'react-bootstrap';
import { Title } from '../../components';
import { AddCountryForm, CountryClockList } from './components';

export const WorldClockPage = () => {
  return (
    <Container>
      <AddCountryForm />
      <CountryClockList />
    </Container>
  );
};
