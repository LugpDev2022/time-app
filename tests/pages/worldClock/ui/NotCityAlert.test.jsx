import { render } from "@testing-library/react";
import { NotCityAlert } from "../../../../src/pages/worldClock/ui";

describe("test on <NotCityAlert />", () => {
  test("should match snapshot", () => {
    const { container } = render(<NotCityAlert />);
    expect(container).toMatchSnapshot();
  });
});
