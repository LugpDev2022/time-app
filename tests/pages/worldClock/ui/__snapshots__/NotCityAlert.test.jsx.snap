// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test on <NotCityAlert /> should match snapshot 1`] = `
<div>
  <div
    class="fade mt-3 bg-info text-dark border-dark border fw-bold alert alert-primary show"
    role="alert"
  >
    Don't leave the city input empty
  </div>
</div>
`;
